### PR TITLE
Fixes #24390 - Content Source is inherited when creating host

### PR DESCRIPTION
**Pull request by dLobatog**
Steps to Reproduce:
1. Create hostgroup with content source
2. Navigate to hosts -> create new host
3. Select hostgroup

Actual results: Content source field is not automatically populated
even though it is provided in hostgroup.

Expected results: Content source field should automatically populate
using value from selected hostgroup.

This is a regression.

### CODE REVIEW

File: app/helpers/katello/hosts_and_hostgroups_helper.rb [hunk 1]
```diff
@@ -97,24 +103,28 @@ def relevant_organizations(host)
...
-      selected_id = fetch_lifecycle_environment(host, options).try(:id)
       orgs = relevant_organizations(host)
       all_options = []
       orgs.each do |org|
-        env_options = ""
-        accessible_lifecycle_environments(org, host).each do |env|
-          selected = selected_id == env.id ? 'selected' : ''
-          env_options << %(<option value="#{env.id}" class="kt-env" #{selected}>#{h(env.name)}</option>)
+        content_object_options = ""
+        accessible_content_objects = if object_type == :lifecycle_environment
```
Comment (jturel) [inline app/helpers/katello/hosts_and_hostgroups_helper.rb]: Can we pass accessible_content_objects array itself into this method rather than object_type?

Comment (jturel) [reply app/helpers/katello/hosts_and_hostgroups_helper.rb]: Good point. Let's leave it as-is!

File: app/helpers/katello/hosts_and_hostgroups_helper.rb [hunk 2]
```diff
@@ -123,6 +133,24 @@ def lifecycle_environment_options(host, options = {})
...
+    def lifecycle_environment_options(host, options = {})
+      content_options(
+        host,
+        fetch_lifecycle_environment(host, options).try(:id),
+        :lifecycle_environment,
+        options
+      )
+    end
+
+    def content_source_options(host, options = {})
```
Comment (jturel) [inline app/helpers/katello/hosts_and_hostgroups_helper.rb]: It'd be nice to have a test which goes over content_options - maybe adding a simple test of content_source_options or lifecycle_environment_options would do the trick

**Comment by dLobatog** [comment]
The 2 failures are `Katello::RedhatExtensionsMediaTest#test_medium_uri_for_no_content_source_or_ks_repo_hg:` and `Katello::RedhatExtensionsMediaTest#test_medium_uri_for_no_content_source_or_ks_repo:`, I don't think they're related though

**Comment by ares** [comment]
@jturel I don't think @dLobatog has merge permissions, if this is acked, could someone merge please? I hope the missing test does not block the merge as I think it's quite high prio

**Comment by jturel** [comment]
@ares ah yes, was waiting on another PR to be merged in order to get the build green. after tests pass I'll merge immediately. thank you!

**Comment by jturel** [comment]
[test katello]

**Pull request merged**